lib/colord/cd-edid.c

```diff
@@ -25,7 +25,9 @@
 #include <string.h>
 #include <glib-object.h>
 #include <math.h>
+#ifdef HAVE_UDEV
 #include <libudev.h>
+#endif
```

Comment:
I do wonder if FreeBSD should just provide a dummy (or functional!) libudev rather than this invasive patch.

Reply:
i.e. there could be a gusb-1.0.pc file in freebsd that works the same as the libusb1.pc one...

Reply:
There was a project called libdevq intended to have a similar interface with udev to ease mesa porting, but yes, there is now a new project called libudev-devd, which implement udev API using FreeBSD devd! https://github.com/FreeBSDDesktop/libudev-devd

Although it is still not used by mesa and Xorg on FreeBSD, it is already packaged so it can be easily installed for testing:
```
$ pkg search udev
libgudev-230_1                 GObject bindings for libudev.
libudev-devd-0.3               libudev-compatible interface for devd
$ pkg-config --modversion libudev
199
$ pkg-config --modversion gudev-1.0
230
```

After installing libudev-devd and libgudev, colord can be built with udev enabled with only two minor issues:

1. `udev_hwdb_*` are not available. Patch to `lib/colord/cd-edid.c` is still required.
2. We still have to skip the `rules` directory because FreeBSD devd and devfs doesn't understand Linux udevd rules.

Reply:
I think that would be a much better thing indeed. Do we have to skip the rules, surely they'll just be installed and not used? A patch for 1. should be pretty easy, and I guess we'll need to add something to the README too.

Reply:
There is currently three udev-related dependencies in `meson.build`:
```meson
gudev = dependency('gudev-1.0')
libudev = dependency('libudev')
udev = dependency('udev')
```
We only have two of them, gudev and libudev, so I think the easiest way to avoid the dependency on udev is to skip the `rules` directory since it is the only place using variables from `udev.pc`.

Reply:
Works for me.